platform/spec/services/connections/start_data_sync_service_spec.rb

```
@@ -0,0 +1,82 @@
+          :default,            # state_machine_name (use default instead of :aasm)
+          ["cannot transition from 'created' to 'running'"] # failures
+        )
+        allow(connection).to receive(:start!).and_raise(error)
+      end
+
+      it "raises the error" do
+        expect { service.call }.to raise_error(AASM::InvalidTransition)
+      end
+
+      it "does not start the workflow" do
+        expect(Temporal).not_to receive(:start_workflow)
```

rubocop (1.68.0)

[convention] Prefer `have_received` for setting message expectations. Setup `Temporal` as a spy using `allow` or `instance_spy`.

(RSpec/MessageSpies)